File: src/models/Adresse.ts
import { Demande } from "./Demande";
import { Utilisateur } from "./Utilisateur";

export class Adresse {


    idAdresse!: number;
  governoate!: string; 
  ville!: string;     
  utilisateurs?: Utilisateur[]; 
  demandes?: Demande[] = [];
  }

File: src/main/java/com/example/backendnourpfe/classes/Adresse.java
package com.example.backendnourpfe.classes;

import com.fasterxml.jackson.annotation.*;
import jakarta.persistence.*;
import lombok.*;

import java.util.List;



@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity

public class Adresse {
     @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
     @JsonProperty("idAdresse")
    private Long idAdresse;
    @JsonProperty("governoate")
    private  String governoate;
    @JsonProperty("ville")
    private String ville;


    @OneToMany(mappedBy = "adressee")

    @JsonIgnoreProperties("adressee")
    private List<Utilisateur> utilisateurs;

@JsonIgnore
    @OneToMany(mappedBy = "adressedemande", cascade = CascadeType.ALL)

    private List<Demande> demandes;
}

